### PR TITLE
Homework 15.02.19

### PR DESCRIPTION
**Pull request by MateAcademy**

### CODE REVIEW

File: src/main/java/com/mateacademy/lessons16/Order.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
+package com.mateacademy.lessons16;
```
Comment (Taniann) [inline src/main/java/com/mateacademy/lessons16/Order.java:1]: this file shouldn`t be in hw pull request

File: src/test/java/com/mateacademy/lessons15/CustomerDaoImplIntegrationTest.java [hunk 2]
```diff
@@ -45,4 +48,20 @@ public void testFindCustomersByCreditLimitMoreThanHundred() throws SQLException
...
     }
 
+    @Test
+    public void testInsertCustomer() throws SQLException {
+        assertTrue(customerDao.insertCustomer(MOST_VALUED_CUSTOMER));
+    }
+
+    @Test
+    public void test2UpdateCustomer() throws SQLException {
+        Customer expected = UPDATE_MOST_VALUED_CUSTOMER;
```
Comment (Taniann) [inline src/test/java/com/mateacademy/lessons15/CustomerDaoImplIntegrationTest.java]: line is redundant

File: src/main/java/com/mateacademy/lessons15/CustomerDaoImpl.java [hunk 3]
```diff
@@ -46,4 +46,63 @@
...
+    @Override
+    public boolean insertCustomer(Customer cast2) throws SQLException {
+        Connection connection = ConnectToDB.getConnection();
+        PreparedStatement statement = connection.prepareStatement("INSERT INTO customers values (?, ?, ?, ?)");
+        statement.setBigDecimal(1, cast2.getCustNum());
+        statement.setString(2, cast2.getCompany());
+        statement.setBigDecimal(3, cast2.getCustRep());
+        statement.setBigDecimal(4, cast2.getCreditLimit());
+
+        int rowsInserted = statement.executeUpdate();
```
Comment (Taniann) [inline src/main/java/com/mateacademy/lessons15/CustomerDaoImpl.java]: you can change it to "boolean isRowInserted = statement.executeUpdate() > 0;" and then close connection and prepared statement
and then just return isRowInserted;
in that way you reduce amount of code

**Review by Taniann — CHANGES_REQUESTED** [review]
add into pull request just hw files: CustomerDao, CustomerDaoImpl, CustomerDaoImplIntegrationTest
all the rest remove from pull request

File: src/test/java/com/mateacademy/lessons16/OrderDaoImplTest.java [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    @Test
+    public void test3deleteOrder() throws SQLException {
+        assertTrue(orderDao.deleteOrder(ORDER.getOrderNum()));
+    }
+
+    @Test
+    public void testinsertOrder() throws SQLException {
+
+        boolean ins = orderDao.insertOrder(ORDER);
+        System.out.println(ins);
```
Comment (SerhiiPasko) [inline src/test/java/com/mateacademy/lessons16/OrderDaoImplTest.java:67]: Why?

File: src/main/java/com/mateacademy/lessons16/Product.java [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+
+import java.math.BigDecimal;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Product generated by hbm2java
+ */
+public class Product  implements java.io.Serializable {
```
Comment (SerhiiPasko) [inline src/main/java/com/mateacademy/lessons16/Product.java:12]: Why I see that in your pull request? In it must be only your homework

File: src/main/java/com/mateacademy/lessons16/ConnectToDB.java [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+import java.sql.SQLException;
+import java.util.Locale;
+
+import com.mchange.v2.c3p0.ComboPooledDataSource;
+import oracle.jdbc.driver.OracleDriver;
+
+/**
+ * @author spasko
+ */
+public class ConnectToDB {
```
Comment (SerhiiPasko) [inline src/main/java/com/mateacademy/lessons16/ConnectToDB.java:15]: this is was on lesson

File: src/main/java/com/mateacademy/lessons15/CustomerDaoImpl.java [hunk 7]
```diff
@@ -46,4 +46,63 @@
...
+        if (rowsInserted > 0) {
+            statement.close();
+            connection.close();
+            return true;
+        }
+        statement.close();
```
Comment (SerhiiPasko) [inline src/main/java/com/mateacademy/lessons15/CustomerDaoImpl.java:77]: close it in finally block. At exeption it will not close

**Review by SerhiiPasko — CHANGES_REQUESTED** [review]
At pull request must be only homework. Also close objects in finally{}

File: src/test/java/com/mateacademy/lessons15/CustomerDaoImplIntegrationTest.java (hunk 2 above)
Comment (MateAcademy) [reply src/test/java/com/mateacademy/lessons15/CustomerDaoImplIntegrationTest.java]: I fixed it

File: src/main/java/com/mateacademy/lessons15/CustomerDaoImpl.java (hunk 3 above)
Comment (MateAcademy) [reply src/main/java/com/mateacademy/lessons15/CustomerDaoImpl.java]: ok, I will do it

File: src/test/java/com/mateacademy/lessons16/OrderDaoImplTest.java (hunk 4 above)
Comment (MateAcademy) [reply src/test/java/com/mateacademy/lessons16/OrderDaoImplTest.java:67]: I see at terminal obdject wich i want to add.

File: src/main/java/com/mateacademy/lessons15/CustomerDaoImpl.java (hunk 7 above)
Comment (SerhiiPasko) [reply src/main/java/com/mateacademy/lessons15/CustomerDaoImpl.java:77]: Don't fixed

File: src/main/java/com/mateacademy/lessons16/Order.java (hunk 1 above)
Comment (SerhiiPasko) [reply src/main/java/com/mateacademy/lessons16/Order.java:1]: Don't fixed

File: src/test/java/com/mateacademy/lessons15/CustomerDaoImplIntegrationTest.java [hunk 8]
```diff
@@ -13,10 +12,14 @@
 public class CustomerDaoImplIntegrationTest {
```
Comment (SerhiiPasko) [inline src/test/java/com/mateacademy/lessons15/CustomerDaoImplIntegrationTest.java:12]: It doesn't work correctly. Start full test class. In result, some of test will be "red"

**Review by SerhiiPasko — CHANGES_REQUESTED** [review]
Read comments

**Pull request closed without merging**